### DISCUSSION TITLE
[Suggestion] Add a Filter Option

### DISCUSSION THREAD

**Discussion by hpacleb** [Forcellrus/Discord-Auto-Publisher#22] [Ideas]
Add a filter option where the user can filter out messages to be automatically published using keywords (and maybe regex if possible?) for a more customizable approach.

**Comment by acehox** [comment]
Such option, among with many more customization were being considered. The goal of the bot is to feature no commands or any kind of a database to maintain a simplistic approach. Maybe in the future I will consider implementing commands and more complex customization features, however it will take me quite some time to implement that and special customization commands would likely be a premium version feature.

**Comment by thewilloftheshadow** [comment]
Maybe something like
`||nopublish||`
that's in the message that the bot checks and then you can edit it out after?

**Comment by acehox** [comment]
I don't think any bot can edit other user's messages. Your approach would require compatibility from both sides. I'll definitely try to make some kind of selective publishing in the future and support customization using RegEx etc. I'm currently in the process of reworking the bot to Eris library as it's more suitable for large scale, as the bot is growing much quicker than I expected.

**Comment by srnyx** [comment]
> I don't think any bot can edit other user's messages. Your approach would require compatibility from both sides.

I think they meant that the user's message would contain a certain keyword which would cause the bot to **not** publish that message. And then the **user** can edit their message to remove that keyword if they wanted to, and the bot would still not publish it (so it only checks the message when it's first sent).

**Example 1:**
User's message:
`@everyone Rules updated, please check #rules! nopublish`

What the bot does:
It **doesn't** publish the message as it contains the keyword `nopublish`
The user can now (if so desired) remove `nopublish` from their message and the bot will still keep the message **un**published.


**Example 2:**
User's message:
`@everyone Rules updated, please check #rules!`

What the bot dies:
The bot **will publish** the message as it **doesn't** contain the keyword `nopublish`

---

I would also like something similar except a user filter on top of a channel filter.
I could set users that the bot would ignore in specific channels.

**Example:**
There is `#announcements` and `#uploads` (both announcement channels)
I send a message in `#uploads` whenever I upload a YouTube video, so I want it to auto-publish when I send a message there. But, sometimes, my admins will make a joke or something in `#uploads`, so I don't want it to auto-publish their messages.
The opposite may go for `#announcements`, where my admins make announcements that should be auto-published but I may side comments that shouldn't be.

**Comment by acehox** [comment]
@srnyx thank you for your input. Something similar to what you suggested has actually been on my list for quite a while now, however I have found very little time to focus on the bot (especially since I make basically nothing from it, so I have to prioritize other projects). At the end, my lack of time has ultimately come out as a good thing since quite a few things changed regarding Discord API in the meanwhile, including new features (such as interactions) which could definitely be implemented for such a purpose. That being said, I aim to do some revamps to the bot in the future with new features, especially now since the bot is on nearly 5.000 servers.

**Comment by Just-Jinn** [comment]
I think if there isnt an @everyone or @silent there is an option not to publish or an option to publish only webhooks/bots
